resolvers/ambulance.js

```diff
@@ -45,41 +45,21 @@ const ambulanceResolvers = {
           return db.ambulance.findByPk(args.id);
         }),
     restoreAmbulance: async (parent, args) => {
-      await Promise.all([
-        db.ambulance.restore({
+      await db.ambulance
+        .findByPk(args.id, { paranoid: false })
+        .then((ambulance) => {
+          if (!ambulance) {
+            throw new Error('Invalid ambulance: ' + args.id);
+          }
+        });
+
+      return db.ambulance
+        .restore({
           where: {
             id: args.id,
           },
-        }),
-        // Restoring event association if event also availiable
-        db.eventAmbulances
-          .findAll({
-            where: {
-              ambulanceId: args.id,
-            },
-            include: [
-              {
-                model: db.event,
-                required: true,
-              },
-            ],
-            paranoid: false,
-          })
-          .then((associatedEvents) =>
-            Promise.all(
-              associatedEvents.map((associatedEvent) =>
-                db.eventAmbulances.restore({
-                  where: {
-                    eventId: associatedEvent.eventId,
-                    ambulanceId: args.id,
-                  },
-                })
-              )
-            )
-          ),
-      ]);
-
-      return db.ambulance.findByPk(args.id);
+        })
+        .then(() => db.ambulance.findByPk(args.id));
     },
```

Comment:
since we're introducing a validation step, i figured it's a lot more readable if we just use `await` instead of promises
```
    restoreAmbulance: async (parent, args) => {
      await validators.validateAmbulance(args.id, true);
      await db.ambulance.restore({
        where: {
          id: args.id,
        },
      });

      return db.ambulance.findByPk(args.id);
    },
```